opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -1263,8 +1281,16 @@ namespace Opm {
             std::vector<int> failed_cells_pb;
             std::vector<int> failed_cells_pd;
 
+            if ( ebosModel.cachedIntensiveQuantities(/*cellIdx=*/0, /*timeIdx=*/0) == nullptr ) {
+                updateSimCache();
+            }
+
             for (int cellIdx = 0; cellIdx < numCells; ++cellIdx) {
-                const auto& intQuants = *ebosModel.cachedIntensiveQuantities(cellIdx, /*timeIdx=*/0);
+                const auto* intQuantsPtr = ebosModel.cachedIntensiveQuantities(cellIdx, /*timeIdx=*/0);
+                assert(intQuantsPtr != nullptr);
+                const auto& intQuants = *intQuantsPtr;
```

Comment:
in the loop-over-all-elements scenario, this would be

```c++
const auto& intQuants = elemCtx.intensiveQuantities(/*spaceIdx=*/0, /*timeIdx=*/0);
```